packages/node-app/src/models/log_data.model.ts

```diff
@@ -0,0 +1,67 @@
+import mongoose, { Schema } from 'mongoose';
+
+import { Unit } from './user.model';
+import { Participant, ParticipantState } from './giveaway.model';
+
+enum Action {
+  CREATE_GIVEAWAY = 'create-giveaway',
+  UPDATE_GIVEAWAY = 'update-giveaway',
+  ADD_PARTICIPANT = 'add-giveaway-participant',
+  UPDATE_PARTICIPANT = 'update-giveaway-participant',
+  GENERATE_WINNERS = 'generate-giveaway-winners',
+}
+
+interface Changes extends mongoose.Document {
+  title?: string;
+  description?: string;
+  prize?: string;
+  rules?: string;
+  image?: string;
+  winners?: string[];
+  participants?: Participant[];
```

Comment:
I think we don't need `participants` entry anymore, just one for a single participant to be used in the update-giveaway-participant action. Also, I believe `winners` is not being used